### ISSUE TITLE
Question about the future; HEVC/AV1

### ISSUE THREAD

**Issue by CristianGrecu** [fgl27/SmartTwitchTV#220]
Hi, as you may know, more and more streamers are testing 4k HEVC streams. Rumors point to additional announcements around Twitchcon US.

I know you've said you don't really want to add support for AV1 since not many devices are capable of hardware decoding. It seems that short term this HEVC codec will be available for everyone to use. Is it even possible to integrate it in this app?

I'm just thinking that most TVs nowdays are 4k so it would be a massive improvement. Leaving resolution aside, I think the other incredible thing is that the bitrate behaves differently, so when a lot of stuff happens on stream, it doesn't become a blurry mess anymore.

We all know that Twitch's official app will be horrible, so if more and more streamers start using 4k HEVC permanently, a regular Desktop PC migh be mandatory for watching on a TV.

Thank you!!

**Comment by fgl27** [comment]
Is possible, but you need a av1 ready device. 

The app will continue to show avc streams as is the only supported by all devices. 

The avc will continue to be used even if the stream is av1, so content will just work

The day this is a real feature I can add a way to detect supported devices.

**Comment by CristianGrecu** [comment]
But what about this HEVC that everyone is testing? If now we are using H.264 and they are testing H.265, shouldn't most devices support it? I assume that the transcoding is still mostly done by twitch unlike AV1 which makes the users handle most of it. 

Also, I thought when a device can't hardware decode it just falls back to software decoding if the CPU is decent. My Sony TV shows AV1 when watching some Youtube 4k videos, but I don't think it has dedicated hardware.

I'm going to be honest, I even love using this app on a tablet, even if it doesn't have regular touch controls. I really hope we won't be left behind with 1080p when even really old smart TVs can handle Youtube 4k (which I think is usually VP9). 

Thanks for answering and sorry if I got anything wrong, I'm not a developer.

**Comment by fgl27** [comment]
Is all the same and part of the same beta program.

If the device support you will be able to use it. 

You can't on most tv devices use se decoder due to the low performance.

You can test it today in the app you can disable the hw decoder and try it 

Most devices struggle with h264 SW decoding so anything more complex is not a option for high resolution, anything over 640p is high.

**Comment by CristianGrecu** [comment]
I see, I guess we'll have to wait and see what happens when they enable it for everybody. Right now they don't even save VODs/clips from these tests, so everything is gone.

When I was watching on a PC, the bitrate was still low, seemed to be 10-12 mbps, but 4K. Codec was showing HEVC and it looked really  nice.

Just one more question cause I don't want to open another thread. I noticed that recently drops started to work perfectly. I am just finding random stuff to claim all the time. Will it ever be possible for this app to properly detect viewing time and collect channel points/receive gifted subs etc.?

**Comment by fgl27** [comment]
On the av1 topic

There is a test hidden stream

If you search for videos "1440p120 av1" it shows. It will only play using avc 1080p on the app, but on twitch site it plays using av1.

https://www.twitch.tv/videos/637388605?t=18h36m31s

I enabled av1 and twitch only sends av1 links not av1 and avc, not a problem is just extra request that will be needed to show it all.

On the 2017 shield it can't play av1 it gets codec errors.

There is a lot of devices like this 

If anyone find a h.265 test vod share the link so I can try it.

**Comment by fgl27** [comment]
Channel points probably not unless twitch changes things today isn't allowed.

Gifts works, as it is given for those logged in the chat, as long you are setup with yours user and chat logging is enabled must work. 

I received a gift once and I only use this app for twitch so it must works for others.

The app can send a on screen warning if you get one.

**Comment by CristianGrecu** [comment]
My bad, I used the wrong wording. Getting a gift if you are targeted works, but you don't get random gifts as a loyalty reward. Back when I was watching from a PS4 in 2021 I used to have like 12 subs for free every month just because the system was seeing me as loyal to some streams. So I would have some sort of priority vs people that watch for 10 minutes. Nowdays 0 random gifts because it's as if I'm not even there.

**Comment by CristianGrecu** [comment]
I don't think there are any VODs for HEVC. Can you do anything so we can test the next time someone is live? Like a hidden setting or something?

Oh btw, only the 4K feed was HEVC, the rest were standard 1080p. Which is another problem in my opinion. I remember when twitch allowed 1440p streams and it would only be 1440p and then straight to 720. It's possible that in the future avc will only get 720p and the rest will be new codecs.

**Comment by CristianGrecu** [comment]
I think I found one. I'm not at the PC right now but it should be 4k HEVC.

https://www.twitch.tv/burkeblack/v/2211849267?sr=a&t=0s

**Comment by fgl27** [comment]
> My bad, I used the wrong wording. Getting a gift if you are targeted works, but you don't get random gifts as a loyalty reward. Back when I was watching from a PS4 in 2021 I used to have like 12 subs for free every month just because the system was seeing me as loyal to some streams. So I would have some sort of priority vs people that watch for a couple of minutes and leave. Nowdays 0 random gifts because it's as if I'm not even there.

I never used an official app, so I'm not aware of all things.

I assume this is also forbidden, as is similar and attached to the same way points are calculated.

Not that is forbidden for say forbidden, is just that isn't allowed an app that will generate points or anything related, as one can use that just to span and get things they don't deserve.

An app using such implementation can be easily banned no questions asked of how and why it does.

The "points" are more than just opening the stream, there are several API requests that need to be made to receive it, I assume the sub and anything related are the same.

If someday that changes we can revise it.

**Comment by fgl27** [comment]
> I think I found one. I'm not at the PC right now to check but live I am 100% sure it was 4k HEVC.
> 
> https://www.twitch.tv/burkeblack/v/2211849267?sr=a&t=0s



> On the av1 topic
> 
> There is a test hidden stream
> 
> If you search for videos "1440p120 av1" it shows. It will only play using avc 1080p on the app, but on twitch site it plays using av1.
> 
> https://www.twitch.tv/videos/637388605?t=18h36m31s

I will use this two VODs to setup something once all is done I share it with you.

thanks!

**Comment by CristianGrecu** [comment]
Thank you!!

**Comment by fgl27** [comment]
I have updated the app to better support those vods, they played just fine but not the best quality out of the box.

in this test apk they do.

[app-debug.zip](https://github.com/user-attachments/files/16461800/app-debug.zip)

install this after extract, this is a testing app it will add a second SmartTwitchTV with a yellow icon.

this app shares no context with the Play Store one you must setup add users etc.

If any issue let me know.

If you see any stream playing AV1 or HEVC let me know here.

Once I improve this version I release it.

**Comment by fgl27** [comment]
I have added this to the settings

![image](https://github.com/user-attachments/assets/07743030-996a-4439-a368-6b3b12203630)

For those two testing VOD it does nothing, disable will not fall back to avc, I assume that isn't ready yet.

Also, no guarantee enabling will make it work even if the streamer is using the codecs

The day we find some live using those we can better test.

**Comment by CristianGrecu** [comment]
I know for a fact that Maximilian_dood has been doing test streams for HEVC 4k and he usually has Twitch staff present. I'll keep an eye out. Other streamers mentioned they want to use 1440p 60 permanently in the future. 

Honestly I don't think anyone is testing AV1 publicly so that might have to wait.

And thanks again for all the work.

**Comment by CristianGrecu** [comment]
That test app seems to work perfectly honestly. I'll test some more on the tv tomorrow because it's getting late. Live will be the true test though.

**Comment by fgl27** [comment]
I have made an updated version

[app-debug.zip](https://github.com/user-attachments/files/16464061/app-debug.zip)

will test this for a few days, test let me know the result

**Comment by fgl27** [comment]
This is the final or almost final version, improve some text and quality selection 

[app-debug.zip](https://github.com/user-attachments/files/16471926/app-debug.zip)

Test this one and let me know

BTW this streamer has HEVC VODs, maybe they do some live in the future for testing

[r0dn3y](https:/twitch.tv/r0dn3y/videos)

**Comment by CristianGrecu** [comment]
I've done some testing. Here's what I found:

I tried a 4k60 HEVC vod, a 1440p60 Elden Ring vod and that AV1.

On the 60hz Sony TV with a gigabit usb network adapter :

4k60 HEVC: seems to work perfectly, 0 skipped frames
1440p60 HEVC: same as above.
AV1: completely broken, blurry and it seems to be on fast forward. After a while graphical artifacts appean on 50% of the screen.

On a 144hz Snapdragon 870 tablet:

4k60 HEVC: almost perfect, like 1 skipped frame. This tablet has a 3k screen I think, so really good overall.
1440p60 HEVC: perfect, no skipped frames, really crisp image.
AV1: no graphical artifacts, still broken. Blurry image and seems on fast forward even though the tablet is 144hz.

Honestly, I wouldn't bother with AV1 until they do public tests, because 120 FPS is too much for normal use anyway.

**Comment by fgl27** [comment]
Av1 is device issue most devices can't play av1
I have a phone with a 888 snapdragon plays fine but no audio. 

Besides that av1 hevc is working great

Will test for a few days then release it

**Comment by CristianGrecu** [comment]
But don't you think the 120 fps is the main issue? Because my TV runs 4k60 on youtube using AV1 without a problem.

**Comment by fgl27** [comment]
You can check the codec capability in settings to know what the tv can and can not

Just because is av1 doesn't mean it's the same thing

There several ways to encode that one stream isn't the standard.

**Comment by CristianGrecu** [comment]
We'll have to see when they start testing 60 fps AV1. That VOD is also quite old so who knows what changed.

It's very nice that HEVC works well, so thank you again. I'll keep looking for a live test but it should be fine.

**Comment by CristianGrecu** [comment]
Btw, unrelated, but have you considered adding an auto catch-up function? Sometimes when Twitch is having a bad day I have to manually get it back to live with the remote (shrinking the red bar as much as possible to the right) to reduce the delay as much as possible. It would be nice if let's say every 30 mins - 1 hour the stream would auto catch-up to live.

**Comment by fgl27** [comment]
there is a player setting that is disabled by default that does that, isn't exported to the user, I'll see to make it configurable

**Comment by CristianGrecu** [comment]
Ok, thank you!!

**Comment by fgl27** [comment]
I enabled the feature to keep the stream position even after buffers and relatedS

file:///home/fgl27/workspace/SmartTwitchTV/apk/app/build/outputs/apk/debug/app-debug.zip

**Comment by CristianGrecu** [comment]
It's usually not a problem, I only mentioned it because yesterday I was having issues with literally 3 streams. In the right corner where it says net speed I usually see 200-300 and the same average. Sometimes just a few streams show 15-20 net speed even if the average is still 200. It is very strange but I think it's Twitch 's fault. Even now Kyedae' s stream does the same: net speed shows 17 and average 200. You can check it if you want, but it's probably an Eu problem.

**Comment by fgl27** [comment]
OK, but that feature is good, I disabled it in the past because was causing issues as it didn't allow configurations but now is working as expected.

So at the start or after a buffer, the player will always try to correct and find the target position.

**Comment by CristianGrecu** [comment]
Yes, of course, I meant I didn't want to bother you with more things!

**Comment by CristianGrecu** [comment]
So I've been using the new app, the latest one you posted, and I think I found a problem. Right after I join a stream I can hear some audio crackles/pops and static. It seems to go away after a while.

I kept going back and forth and the released app doesn't do this. Was there anything changed related to audio? Or could it be the auto catch-up thing misbehaving?

**Comment by fgl27** [comment]
I didn't notice it, maybe is device-related...

The catch-up happens every time that the latency is off.

At the beginning, you can see the latency value (top right) going down or up depending on how the playback started.

I made two testing app tests to see if any diff, test let me know.

I slowed down the catch-up to see if was it now you may not notice it.

[app-debugs.zip](https://github.com/user-attachments/files/16480929/app-debugs.zip)

**Comment by CristianGrecu** [comment]
I tested it on the tablet and also using some Beyerdynamic dt770 and an amp on the TV, so I don't think it's device related because the old one doesn't do it.

I'll test with this one you posted and I'll be back asap.

**Comment by CristianGrecu** [comment]
So those 2 apps that you posted seem to be fine. I swear only the one above from 18 hours ago is the one that does it. Quite noticeable pops and crackles, I got it to do it on 7 different streams. Could it be a bug with that apk only?

**Comment by fgl27** [comment]
I slowdown the catch up system that is the only diff

Just test that one if you notice any issue let me know. 

Thank you!

**Comment by CristianGrecu** [comment]
So I tested the latest app with catch-up. Optical into amplifier and DT770 and it still happens. I tried multiple streams including clean music so I can hear the background. The old app sounds really clean but with the new one there's constant pops but lower volume than the content itself.

**Comment by fgl27** [comment]
You can look at the top right corner latency to broadcast it get lower and lower until reach the expected value

2.5 or 4.5 seconds is the expected. 

Once the value gets to that value is when the issue stops?

The app without catch has the issue? I shared two apps on last post, which has the issue?

**Comment by CristianGrecu** [comment]
Yes I was comparing the current one with the latest catch-up, because I wanted to switch between them (can't have 2 new yellow ones installed). But my latency doesn't slowly go lower like you said. Just now I clicked on a stream and it just sits at 1.30. Others just start at 2.7.

Btw, I have the buffer size at 0.1 in the settings. This is how I've been using it for years and I had no problems.

**Comment by fgl27** [comment]
What I need to know is if the one that doesn't have catch on the name has the issue. 

Because if it does the problems is something else

**Comment by CristianGrecu** [comment]
I'm trying the one without catch now and I can't get it to do it at all, sound is very good. The one from 18 hours ago is the worst one, loud enough crackle to hear it even through speakers.

**Comment by fgl27** [comment]
OK I'll try another version later, one that allows to disable/enable the catch 

At the same time slow down even more then we test again

Thank you for the help, great work, by myself I would not notice that issue.

**Comment by CristianGrecu** [comment]
No problem, I also kept looking for an HEVC live but I couldn't find any so that will have to wait. Other than that it seems to work really well.

**Comment by fgl27** [comment]
[app-debug.zip](https://github.com/user-attachments/files/16484437/app-debug.zip)

this has the option in setting making easier to test for audio issue.

**Comment by CristianGrecu** [comment]
Ok, so you can still independently have low latency with or without the catch up, right? But disabling just low latency is faster for testing if I understand correctly. 

I'll check and I'll tell you if I find anything that doesn't work. Thank you.

**Comment by fgl27** [comment]
yes, you can leave the setting option 'catch-up low latency' disabled, and low latency will work.

and if you enabled 'catch-up low latency', and wanna disable it from the player just disable low latency and all will be off, for easier and faster testing

**Comment by CristianGrecu** [comment]
So I've been testing the app for a while. From what I can tell the audioo issues are very minimal.

Unfortunately the catch up feature doesn't really fix what I was thinking of. I'll try to explain better.

So let's say I have stream A open on the tv. Using the remote I can push that white dot above the pause all the way to the right, bringing the delay as low as possible. Now let's say I leave it like that. After a while, even without a single buffer, the delay will very slowly drift back. So 0 skipped frames and no buffer and I find that streams still lose 1-1.5 secs (vs the same app that was just opened) every hour to few hours. So the same app newly opened will be slightly ahead. 

That's why I thought you could literally add an automatic remote input that moves that white dot to the right every 30 mins or so.

I'm sure the catch up function works if there are actual buffers which is great, but it doesn't help with the slow, subtle drift that happens. I hope I'm nit nitpicking, I'm sure many people don't care, but it's nice to be as close to actually live as possible.

But for now don't worry about it, it is really good. Maybr keep the catch up as a toggle in the settings just to be safe.

**Comment by CristianGrecu** [comment]
I've testing some more and it seems to maintain the delay better than the old one, so I think I was wrong. And sorry for the wall of text above.

**Comment by fgl27** [comment]
don't worry

latency depends on so many variables, a simple thing as your device clock being a little off in relation to the server clock or streamer can make it go up/down

Android OS and clock sync is something that is very often not perfect.

**Comment by CristianGrecu** [comment]
Oh, I see. There's still that one problem left, but it's also in the old app. Just a few streams every single day are always on the edge of buffering. 

Sometimes it's a single stream out of many. I can tell because up in the right corner the net speed shows 15-17 instead of 200+ and those streams are very sensitive. After a bit the net speed dips to 6, it buffers and the goes back to 15-17.

I tried an unlimited 4G sim card so I know it's not my internet/DNS. It also doesn't happen on PC. I still don't think there's anything you can do but I wanted to let you know.

Unless it gets worse, it is what it is. Those streams are still watchable on source, but they are on the edge at all times.

**Comment by fgl27** [comment]
the app will use all available bandwidth, doesn't matter what your ISP is the limit is most of the time is on the Twitch server.

So using auto-quality playback is the best option to avoid buffers as it will always lower the quality buffer, if that doesn't work is because the internet stops not a bandwidth issue but a connection issue

**Comment by CristianGrecu** [comment]
Yes, but don't understand how 50 streams show 200+ and work flawlessly, and just 1 or 2 every day are always struggling. This has to be on Twitch's side, right?

It only started happening recently, before I've never seen that happen. Some streams show 300-400 in the right corner.

It's also weird that it doesn't happen in Chrome. For example if you have a few minutes to check, Maya's stream does it for me right now. 15-17 net activity. It's also not because it's IRL, desktop streams do it too.

**Comment by fgl27** [comment]
twitch site doesn't tell you how much bandwidth is using

windows tells me Maya is about 8 to 9 and lirik 11 to 12

on my app may is the same lirik is 300+

so the website doesn't use all the available bandwidth

**Comment by CristianGrecu** [comment]
Yes Lirik shows 200+ for me too in the app. Problem is those streams that show 15 instead of hundreds are very sensitive. I've only been watching on source for years, I don't want to go below that.

Oh well, at least I know that it does the same on your side, so thank you for testing. I appreciate it!

**Comment by fgl27** [comment]
Yes I remember something 

Twitch does limit the bandwidth in e very specific way

Example if bitrate is 10Mbps and the stream is segmented in 2 seconds 

Meas that in the net activity it will show 20Mb, because that is the size of what is being download 

2s x 10Mb = 20Mbps

I notice that sometimes the server will limit the bandwidth in that value allowing one to play but is at the limit

The website player will only play and download a single quality but this app in auto mode will play one and download two

So in case it lags it can switch to the lowest quality 

But when the server is limiting that isn't enough do download two quality does the app lags and buffers 

In this scenario to validate you can switch from auto toml.source and the problem of buffer must be gone, the speed will be enough to play that one quality 

Do you use auto or source?

In that one senário source is the best

I do recommend auto but isn't for every senário

**Comment by CristianGrecu** [comment]
The first thing I did after you gave me the app was to go in the settings and do these things: Default quality: Source, Buffer Size 0.1 to all 3 things. This is how I've been using it for years. I don't even know how 720p looks like.

**Comment by fgl27** [comment]
Source is the only way to minimize any buffer lag but you don't need to change the buffer value lowest value is the best

Twitch server are limiting because they are on the limit in that senário the best way to avoid buffers is to disable low latency as any server lag will cause a buffer of the buffer is to small 

And this buffer has no relation to the one on settings, you can't control total buffer size it is based on the latency for lives streams

**Comment by CristianGrecu** [comment]
So 0.1 start buffer in the settings is the best, right?

**Comment by fgl27** [comment]
0.1 is the best. 

If there is a limit that is on you IP.

Not the device, if you play the same stream on two devices one may be affected one not.

You say it never buffers but have you tried those stream that are limited? How do you know is limited only if you use this app.

Nothing the app can do if there is a limit. It can't make the correction better if you are using the same connection else where something got to give.

**Comment by fgl27** [comment]
Also isn't the website better then the app, but the device

At least he computer is probably 10xm better then my Android tv

Have you tried the website on yours Android devices, I doubt that it will be a gone or similar experience as on the computer

Computer are way more powerful this days, and due to it's performance it will be able to avoid a lag much easier then a Android device.

**Comment by CristianGrecu** [comment]
So if the top right corner says 200-400 I can leave that stream for 10 hours, no buffer and almost same delay as it was at the beginning.

Those that show 17 net activity are very unstable but only in the app. I can watch on a Windows PC and I can't even tell which ones have problems in the app.

This started recently and a stream that has the issue on Monday might be fine on Tuesday.

**Comment by fgl27** [comment]
Se my post open the website on the Android device on those limited streams, you will see isn't a app issue. 

Is just there computer performance avoiding a lag that the Android low performance can't avoid

**Comment by CristianGrecu** [comment]
I understand what you are saying, but why 3 streams out of 50? Why were these streams perfectly fine a week ago?

I don't think the device has anything to do with this. For example the worst I've found today is "Berks" if you can search it in the app. I left his stream open for hours last week, no issue. Today it's showing 17-20 net speed and it buffers a lot.

**Comment by fgl27** [comment]
Then now open the website on that stream on the Android device and tell me the result

**Comment by CristianGrecu** [comment]
It seems to have some issues in the browser too, but why a 6 mbps stream that has nothing fancy.

**Comment by CristianGrecu** [comment]
Wait a minute, I was using Opera, my bad. With Chrome on my tablet the stream that is broken in the app works perfectly.

**Comment by CristianGrecu** [comment]
If you have a few free minutes can you please just test "Berks" and tell me if it works perfectly or if it shows low net speed like Maya's and buffers?

**Comment by fgl27** [comment]
from my experience is like a season, at least for me has always been like this, from nowhere some streams are like that then after some time all OK.

that is why I have added 3 levels of low latency, as my internet is NOK sometimes the best is to let low latency disabled, the buffer is around 14s, and in most cases, if any lags occur will recover in those 14s, I set chat delay to latency to broadcast and that is it

It is not ideal but for me works.

by what you describe I assume Twitch player has a way to detect low bandwidth and just disable low latency to avoid re-buffer by letting the buffer be more than the normal

anyway maybe in the future I come up with the same, isn't as simple to have a tool to detect a possible re-buffer but is something that can be done.

anyway keep investigating if you find something share it, is best to have a discussion than to just expect that is all ok all the time

from this discussion with you we have improved a few things which is great!

**Comment by CristianGrecu** [comment]
Honestly I don't know if you'll believe me, but there was a time during 2022-2023 using your app and I never had to worry about anything. Always source, always low delay.

My biggest fear is that they are cutting costs more and more. I'm from Romania so at some point they might say: well those people are spending thousands on bits/subs etc. like americans , let's cut their low latency, maybe force them to not watch on Source. At this point I'm expecting anything.

**Comment by CristianGrecu** [comment]
I've done more testing. This is so frustrating because I'm so tired of doing QA testing for these massive corporations.

Basically streams that fall behind using the app show 0 skipped frames and low net speed in the upper right corner.

The exact same streams eventually fall behind on PC using Chrome.

Streams that are fine using the app are also fine on PC.

I also tried using a simcard and same thing happens. Streams that are fine show 80 mbps net speed, ones that are affectef show 15-17 which drops to 5-6 causing buffer.

So the app works perfectly fine. I'm 99% sure it's not my ISP. If you want to help me confirm this is global, let me know and let's test tomorrow. I'll give you like 5 streams that are bad and you can confirm that we have the same net speed, and maybe 5 that are fine. If we match 10/10 we at least know it's global.

I've already opened a ticket with Twitch, we'll see how that goes.

**Comment by fgl27** [comment]
Yes sure you can share some so we can test.

for me it has been like that, as I wrote "seasons", sometimes is fine at 200+ net speed, then some days 100+ for weeks then from nowhere random streams are barely playing source quality...

After tuning the latency on the app I don't notice any issue, so maybe the issue for me is more often I tune the app to avoid it altogether.

let's see what they reply.

**Comment by CristianGrecu** [comment]
So, just for today:

These 5 show 100-400 mbps net speed and the usual 17 mbps activity. No buffers, no delay gain:

Lirik
MissMikkaa
DatModz
AnnieFuchsia
Nugiyen
Insym

Now these streams are the ones that are bad today, should show an immediate drop in net speed (NOT activity) to about 16 mbps jumping to 45 and back to 10-16, drops to 3-6 eventually and should gain delay:

PirateSoftware
DansGaming
nl_Kripp
Towelliee
Ezekiel_iii
Pohx

Let me know, thank you!!

**Comment by fgl27** [comment]
Yes that is the same to me, the ones above fine 100+ speed some tunes 400+

the ones at the bottom are playing at bare minimum, if stream bitrate is 7mb the speed is 14.

As I explain before 2 seconds segmented streams will play like that when limited.

**Comment by CristianGrecu** [comment]
Ok, thank you so much!!

Now we know it's not the app because I did PC testing, and it's not my internet.

Why do you think nobody says anything in ny of the chats? Do people just accept the eventual 20 sec delay, do they pause/unpause and move on? Very weird to me.

**Comment by CristianGrecu** [comment]
Btw, for these broken streams the auto catch up doesn't do anything. I found streams with 8 sec delay just sitting like that.

I usually just click the low latency setting (to refresh the stream) or quickly move the dot on the player to the right. Could you do something so the app does it automatically every 20-30 mins?

Like a literal press of the low latency setting which refreshes the stream. Or anything that copies a remote input, nothing fancy.

**Comment by fgl27** [comment]
check if is enabled it must work, works for me on any stream, it will only not do its thing if you change the position using the progressbar

t changes at most 500ms each minute so it can take a while.

**Comment by CristianGrecu** [comment]
I think it's working for the good streams, you're saying it should work even if it buffered a lot? I found some streams with the progress white dot almost in the middle and a bunch of red to the right.

I have the latest version with everything enabled.

**Comment by fgl27** [comment]
if you have not change using the progress bar yes

but if you use the progress bar no it stops as it assumes you have a new manual position

so have you use the progress bar to change the position?

**Comment by CristianGrecu** [comment]
You know what, I think yes, I've been doing it a lot. The second I see a buffer It annoys me and I drag the bar to the right. I'll leave it alone, sorry.

**Comment by fgl27** [comment]
OK expect frustration not a problem

Try and let me know

That progress bar feature is to prevent a very normal use case that you may wanna to rewind but the player keeps pushing it forward

Today there is no logic to disable it.

**Comment by CristianGrecu** [comment]
I usually use it to lower the delay, very rarely to rewind. But I'll let it do what it's supposed to. Thanks.

**Comment by fgl27** [comment]
I push this changes to the play store app, you can use that one for now

any improve we do on a future update

**Comment by CristianGrecu** [comment]
Sure, thank you. Also all of the streams seem to be good today, no more low net speed.

I don't know if it has anything to do with the ticket that I've opened. Didn't even get an answer. Oh well.

**Comment by fgl27** [comment]
great, I assume not related, is just server load, if after some time too many users start watching the server will load and cause the issue.

but may not 🤞

**Comment by CristianGrecu** [comment]
Ok, I'm not going to close the issue, it's up to you!

Thank you for everything.

**Comment by fgl27** [comment]
OK, update with Twitch reply if they do, then we can close this.

**Comment by CristianGrecu** [comment]
Absolutely no reply and streams have been working perfectly.

It's up to you, but because I was having the same issues on PC, it has nothing to do with this app.

**Comment by CristianGrecu** [comment]
I spoke too soon. The net speed is really high on all streams, no skipped frames, but streams still load with a higher delay. Also they seem to gain delay over time with 0 buffering. 

Last night it wasn't doing this, definitely a different problem. It's like something is wrong with the low latency mode. 

I guess it's still Twitch, butI don't have time to check right now.

**Comment by CristianGrecu** [comment]
It had been working well. Also thank you for the new interface changes. My tv shows AV1 hardware encoder enabled, max 4k 66 fps. That would explain why that vod was completely broken, the 120 fps was too much. Also shows a hardware encoder for HEVC 4k 129 fps.

We'll see when live tests actually start.

**Comment by fgl27** [comment]
I don't think that vod has audio at all, none of my devices reproduce it even on a computer.

Do you have a computer that has audio?

**Comment by CristianGrecu** [comment]
Yes, you are absolutely right. I was mostly testing the app, it's been months since I last checked it on PC.

**Comment by CristianGrecu** [comment]
In case you haven't watched the opening ceremony for Twitch con, they have officially announced HEVC for early 2025.

The stream was 1440p60 and it worked perfectly on my TV, but I do have a question. I know you don't officially support mobile devices but the stream refused to play on my tablet. It said player error 4003, which I think has something to do with the web socket. I just checked a 1440p HEVC VOD and it works perfectly on the same tablet, 0 skipped frames.

It's not a big deal, hopefully it will work fine on most devices, there's plenty of time until 2025.

**Comment by CristianGrecu** [comment]
But that can't be right. It says in the settings I have a HEVC hardware decoder for 4k 240fps and 8k 60fps.

Also Maximilian_dood's vod for Elden ring works perfectly, both say HEVC in the top right corner. Are there multiple versions of HEVC?

**Comment by CristianGrecu** [comment]
OK so I checked on PC, both vods are exactly the same HEVC version, same backend version as well. It's so weird that one works and one doesn't.

**Comment by CristianGrecu** [comment]
Yeah, the official Twitch mobile app allows me to select 1440p and that's also broken, I get stutters and only sound. It's broken on my phone as well. 

I think there's something wrong with this particular stream. I'm sorry that I bothered you.

**Comment by CristianGrecu** [comment]
Just one more question if you don't mind. I found a really old Xiaomi MI A1 from 2017 that I still had and decided to give it a try. This vod from last night worked perfectly with 1440p HEVC and this phone is super weak with a snapdragon 625

You know the android ecosystem pretty well. Do you have any idea why the same tablet that plays the 4K vod would be the only device I have doing this?

If not I'll just have to see if I can find another live HEVC later. It's just that it's so strange that I want to know why out of curiosity at this point.

Thanks!

**Comment by fgl27** [comment]
Seems to be a QCOM issue my 888 device has the same problem. 

Even the codec saying it's the same in the site must have a diff on a computer is possible to check

Next time I'm on the computer I check

As most Android devices today are QCOM there is probably a lot of people with the same issue, unfortunately Twitch isn't the company that cares, they probably didn't even test this codec.

**Comment by CristianGrecu** [comment]
Sure, but the thing is, the 1440p Elden Ring vod has the exact same codec number as the vod from yesterday. I checked every digit.

That Elden Ring one and the 4K one that I linked you with Fallout London both work on this tablet.

Here's my guess why. I might be wrong, but I don't think those hevc tests were included in the mobile app. Last night they probably did the first hevc test in the official app as well as desktop, which would explain why it's so broken.

**Comment by CristianGrecu** [comment]
So I found another 1440p hevc stream, this time IRL (EsfandTV). This one works perfectly everywhere. It's even in the mobile app. So that Twitch one was probably an exception and not the rule.

**Comment by fgl27** [comment]
OK next time I'm in a computer I use those two as examples to see what the diff is.

**Comment by fgl27** [comment]
It doesn't seem to have any diff between the streams other than Twitch has subtitles.

I don't know if that breaks it, when I have more time will see if there is anything that can be done from the app side to finger this out.

**Comment by CristianGrecu** [comment]
Yep, good news is that it was stuttering badly in the official mobile app too (no error code) , so a lot of people will report it.

Also, if you update the app in future will you enable hevc by default? Because I don't think most people look through the settings etc.

**Comment by fgl27** [comment]
Once I have time to address the error codes issues I'll change the codec to default enabled.

**Comment by lextra2** [comment]
@fgl27 

Not sure if this was talked about already, but some streamers who use HEVC have no codec type shown in the overlay

https://twitch.tv/klean

![IMG_20241230_175055](https://github.com/user-attachments/assets/4e025af9-6b92-45df-92c1-3c90035fffe8)

**Comment by fgl27** [comment]
No, not yet...

maybe the information is not available or in a unexpected format.

Will check thanks!

**Comment by CristianGrecu** [comment]
Many streamers use the old codecs with 1440p60, you can easily tell if the quality drops right to 720p without the option to select 1080p. New HEVC streams should be able to revert to 1080p AVC etc.

**Comment by fgl27** [comment]
Twitch changed part of the implementation and the app doesn't know what to show, so it shows blank

For auto playback is fine but for source isn't.

I will need to push an APK update to fix this I will in a few days.
